packaging/os/pulp_repo.py

```diff
@@ -0,0 +1,280 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2016, Joe Adams <adams10301@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: pulp_repo
+author: "Joe Adams <adams10301@gmail.com> github.com/sysadmind"
+short_description: Add or remove Pulp repos from a remote host.
+description:
+   - Add or remove Pulp repos from a remote host.
+version_added: "2.1"
+requirements: []
+options:
+  name:
+    description:
+      - name of the repo to add or remove (correlates to pulp's repo-id)
+    required: true
+    default: null
+  state:
+    description:
+      - The repo state
+    required: false
+    default: present
+    choices: [ "present", "absent" ]
+  repo_type:
+    description:
+      - Repo plugin type to use (i.e. rpm, docker)
+    required: true
+    default: null
+  login_user:
+    description:
+      - User to connect to the pulp server as.
+    required: false
+    default: null
+  login_password:
+    description:
+      - Password to connect to the pulp server with.
+    required: false
+    default: null
+  feed_url:
+    description:
+      - Upstream feed URL to receive updates from.
+    required: false
+    default: null
+  relative_url:
+    description:
+      - Relative url for the local repository.
+    required: false
+    default: null
+'''
+
+EXAMPLES = '''
+# Create a new repo with name 'my_repo'
+- pulp_repo: name=my_repo state=present
+
+# Create a repo with a feed and a relative URL
+- pulp_repo: name=my_centos_updates feed_url='http://mirror.centos.org/centos/6/updates/x86_64/' relative_url='centos/6/updates' state=present
+
+# Remove a repo from the pulp server
+- pulp_repo: name=my_old_repo state=absent
+'''
+
+RETURN = '''
+repo: Name of the repo that the action was performed on.
```

Comment:
please have a look in other modules how to specify returns. e.g. https://github.com/ansible/ansible-modules-extras/blob/devel/cloud/cloudstack/cs_account.py